app/src/main/java/info/nightscout/android/medtronic/service/MedtronicCnlIntentService.java

```diff
@@ -287,11 +286,11 @@ protected void onHandleIntent(Intent intent) {
             } catch (UnexpectedMessageException e) {
                 Log.e(TAG, "Unexpected Message", e);
                 sendStatus("Communication Error: " + e.getMessage());
-                pollInterval = MainActivity.pollInterval / (MainActivity.reducePollOnPumpAway?2L:1L);
+                pollInterval = 60000L; // retry once during this poll period
```

Comment:
There was a single retry previously for time and status as part of cnl reader but remove those as potential problems when signal is weak. Comms errors here are usually due to intermittent radio noise and a single retry after 60 sec catches these and reduces gaps. It doesn't repeat if a user is away but does catch noisey comms and is much safer then before.